Projects/TravelCertificate/Sources/Views/WhoIsTravelingScreen.swift

```diff
@@ -7,117 +9,85 @@ import hCoreUI
-                                        ?? 0) && index == coinsuredMembers.count - 1
-                                    {
-                                        addPeopleButton
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-                .disableOn(TravelInsuranceStore.self, [.postTravelInsurance])
+        ) { state in
+            let travelInsuranceConfig = state.travelInsuranceConfig
+            CheckboxPickerScreen<CoInsuredModel>(
+                items: {
+                    let contractStore: ContractStore = globalPresentableStoreContainer.get()
+                    let contract = contractStore.state.contractForId(travelInsuranceConfig?.contractId ?? "")
 
-                hSection {
-                    InfoCard(text: L10n.TravelCertificate.whoIsTravelingInfo, type: .info)
-                }
-                hSection {
-                    hButton.LargeButton(type: .primary) {
-                        vm.validateAndSubmit()
-                    } content: {
-                        hText(L10n.General.submit)
-                    }
-                }
-                .padding(.bottom, 16)
-                .trackLoading(TravelInsuranceStore.self, action: .postTravelInsurance)
-            }
-        }
-        .presentableStoreLensAnimation(.default)
+                    let insuranceHolder = CoInsuredModel(
+                        firstName: state.travelInsuranceModel?.fullName,
+                        SSN: contract?.ssn,
+                        needsMissingInfo: false
+                    )
+                    var allValues = [(object: insuranceHolder, displayName: insuranceHolder.firstName ?? "")]
+                    let allCoInsuredOnContract =
+                        contract?.coInsured.filter({ !$0.hasMissingInfo })
+                        .map { (object: $0, displayName: $0.fullName ?? "") } ?? []
+                    allValues.append(contentsOf: allCoInsuredOnContract)
+                    return allValues
+                }(),
+                preSelectedItems: {
+                    let contractStore: ContractStore = globalPresentableStoreContainer.get()
+                    let insuranceHolder = CoInsuredModel(
+                        firstName: state.travelInsuranceModel?.fullName,
+                        SSN: contractStore.state.contractForId(travelInsuranceConfig?.contractId ?? "")?.ssn,
+                        needsMissingInfo: false
+                    )
+                    return [insuranceHolder]
+                },
+                onSelected: { selectedCoInsured in
+                    let store: TravelInsuranceStore = globalPresentableStoreContainer.get()
 
-    }
+                    selectedCoInsured.forEach { coInsured in
+                        let contractStore: ContractStore = globalPresentableStoreContainer.get()
```

Comment:
can we rework this?
I think it would be best to send a list to the store where it will handle logic 
maybe rename
case setPolicyCoInsured(PolicyCoinsuredPersonModel)
to 
case setPolicyMembers([PolicyCoinsuredPersonModel])
and inside store, you can check logic what should be set where.

Reply:
Yeah I thought about this as well. Will change it!